### PR TITLE
Remove Unused AddSequencedLeaves from LogTX interface

### PR DESCRIPTION
**Pull request by gdbelvin**
- [x] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.

### CODE REVIEW

File: storage/log_storage.go [hunk 1]
```diff
@@ -100,12 +100,6 @@ type LogTreeTX interface {
 	// employing this property. Consult the call sites of this method to be sure.
 	DequeueLeaves(ctx context.Context, limit int, cutoff time.Time) ([]*trillian.LogLeaf, error)
 
-	// AddSequencedLeaves stores the passed in leaves at the log positions
-	// specified in the `LeafIndex` field. The indices must be contiguous.
-	//
-	// See LogStorage.AddSequencedLeaves comment for more details.
-	AddSequencedLeaves(ctx context.Context, leaves []*trillian.LogLeaf, timestamp time.Time) ([]*trillian.QueuedLogLeaf, error)
```
Comment (Martin2112) [inline storage/log_storage.go:107]: All the implementations have been cleaned up?

File: storage/postgres/log_storage_test.go [hunk 2]
```diff
@@ -373,13 +373,12 @@ func initAddSequencedLeavesTest(t *testing.T) addSequencedLeavesTest {
...
-			t.t.Fatalf("Failed to add sequenced leaves: %v", err)
-		}
-		// TODO(pavelkalinnikov): Verify returned status for each leaf.
-		return nil
-	})
+	ctx := context.Background()
+	if _, err := t.s.AddSequencedLeaves(ctx, t.tree, leaves, fakeQueueTime); err != nil {
+		t.t.Fatalf("Failed to add sequenced leaves: %v", err)
+	}
+
```
Comment (pavelkalinnikov) [inline storage/postgres/log_storage_test.go:380]: Could you remove this blank line?

File: storage/log_storage.go (hunk 1 above)
Comment (pavelkalinnikov) [reply storage/log_storage.go:107]: MySQL?
In-memory storage impl? (not sure it still exists)
Mocks?

Comment (pavelkalinnikov) [reply storage/log_storage.go:107]: The MySQL one is still used but only internally. Could you unexport it as `addSequencedLeaves`?

Comment (gdbelvin) [reply storage/log_storage.go:107]: No implementions use this now

File: storage/postgres/log_storage_test.go (hunk 2 above)
Comment (gdbelvin) [reply storage/postgres/log_storage_test.go]: done

File: storage/log_storage.go (hunk 1 above)
Comment (Martin2112) [reply storage/log_storage.go:107]: It looks like the MySQL one might still exist? Anyway it can be sorted out later.

**Comment by gdbelvin** [comment]
It only exists as a private method

**Comment by pavelkalinnikov** [comment]
It's still [exported](https://github.com/google/trillian/blob/e66c3f708182ff36c01864e45677058b7bf025ea/storage/mysql/log_storage.go#L564), so not really a "private" one. I suggested unexporting it, i.e. renaming to `addSequecedLeaves`.

**Comment by gdbelvin** [comment]
True, but because it's on a private struct, and because it only gets
returned through the storage.ReadOnlyLogTX interface, I'm not sure an
external caller can get to it.

**Pull request merged**